TM1py/Services/CellService.py

```diff
@@ -1149,6 +1146,68 @@ def _build_blob_write_process(self, cube_name: str, process_name: str, blob_file
 
         return dataload_process
 
+    def _build_cube_to_blob_process(self, cube: str, variables: List[str], skip_variables: List[str], top: int,
+                                    skip: int, skip_zeros: bool, skip_consolidated_cells: bool,
+                                    skip_rule_derived_cells: bool, value_separator: str, process_name: str,
+                                    view_name: str, file_name: str, header_line: str,
+                                    quote_character: str = '"', sandbox_name: str = None) -> Process:
+        process = Process(
+            name=process_name,
+            datasource_type='TM1CubeView',
+            datasource_view=view_name,
+            datasource_data_source_name_for_server=cube,
+            datasource_data_source_name_for_client=cube,
+            datasource_ascii_delimiter_char=value_separator,
+            datasource_ascii_decimal_separator='.',
+            datasource_ascii_thousand_separator='')
+
+        # Create variables in process data source as all String
+        for variable in variables:
+            process.add_variable(name=variable, variable_type='String')
+
+        prolog_procedure = f"""
+        SetOutputCharacterSet('{file_name}.blb','TM1CS_UTF8');
+        {self.generate_enable_sandbox_ti(sandbox_name)}
+        ViewExtractSkipCalcsSet('{cube}', '{view_name}', {'1' if skip_consolidated_cells else '0'});
+        ViewExtractSkipRuleValuesSet('{cube}', '{view_name}', {'1' if skip_rule_derived_cells else '0'});
+        ViewExtractSkipZeroesSet('{cube}', '{view_name}', {'1' if skip_zeros else '0'});
+        DatasourceAsciiDelimiter='{value_separator}';
+        DatasourceAsciiQuoteCharacter='{quote_character}';
+        nRecord=0;
+        """
+        process.prolog_procedure = prolog_procedure
+
+        # ignore some variable in file and output variables ordered by their ordinal e.g. v2,v4,v5,v11
+        comma_sep_variables = ",".join(sorted(set(variables) - set(skip_variables), key=lambda v: int(v[1:])))
+        data_procedure_pre = """
+        nRecord = nRecord + 1;
+        """
+        if header_line:
+            data_procedure_pre += f"""
+            IF ( nRecord = 1);
+              TextOutput('{file_name}.blb',{header_line});
+            ENDIF;
+            """
+        if top:
+            data_procedure_pre += f"""
+            IF (nRecord > {top});
+              ProcessBreak;
+            ENDIF;
+            """
+        if skip:
+            data_procedure_pre += f"""
+            IF (nRecord <= {skip});
+              ItemSkip;
+            ENDIF;
+            """
```

Comment:
Is the source view of this process the full view e.g isn't filtered by top / skip? If so, i'm not sure this logic is going to work if both `top` and `skip` arguments are used. 

Let's say for example that `top=10` and `skip=5`. Using the above logic, `nRecord` would be incremented for every row so only the first 10 records would be exported to the blob before ProcessBreak is called. I would expect the  first 5 records to be skipped then records 6-15 to be exported to the blob? It might be worth having a test with this scenario to verify the behaviour?

Reply:
Thanks for the catch. Resoved with [6b3b0d8](https://github.com/cubewise-code/tm1py/pull/885/commits/6b3b0d899a674cb00d493f83cd542a46809413b5)